go/teams/proofs.go

```diff
@@ -86,3 +90,58 @@ func (p *proofSetT) AddNeededHappensBeforeProof(a proofTerm, b proofTerm) *proof
 	p.proofs[idx] = append(p.proofs[idx], proof{a, b})
 	return p
 }
+
+// lookupMerkleTreeChain loads the path up to the merkle tree and back down that corresponds
+// to this proof. It will contact the API server.  Returns the sigchain tail on success.
+func (p proof) lookupMerkleTreeChain(ctx context.Context, g *libkb.GlobalContext) (ret *libkb.MerkleTriple, err error) {
+	leaf, err := g.MerkleClient.LookupLeafAtHashMeta(ctx, p.a.leafID, p.b.sigMeta.PrevMerkleRootSigned.HashMeta)
+	if err != nil {
+		return nil, err
+	}
+	if p.a.leafID.IsUser() {
+		ret = leaf.Public
+	} else {
+		ret = leaf.Private
+	}
+	return ret, nil
+}
+
+// check a single proof. Call to the merkle API enddpoint, and then ensure that the
+// data that comes back fits the proof and previously checked sighcain links.
+func (p proof) check(ctx context.Context, g *libkb.GlobalContext) error {
+	triple, err := p.lookupMerkleTreeChain(ctx, g)
+	if err != nil {
+		return err
+	}
+	laterSeqno := triple.Seqno
+	earlierSeqno := p.a.sigMeta.SigChainLocation.Seqno
+	if earlierSeqno > laterSeqno {
```

Comment:
`p.a.sigMeta.SigChainLocation.Seqno` is a merkle seqno?
It's from https://github.com/keybase/client/blob/master/go/teams/chain_parse.go#L129 though